app/controllers/PromoLandingPage.scala

```diff
@@ -23,102 +24,111 @@ import utils.Tracking.internalCampaignCode
 import views.html.promotion._
 import views.html.weekly.landing_description
 import views.support.PegdownMarkdownRenderer
-import play.api.libs.concurrent.Execution.Implicits.defaultContext
-import play.api.libs.ws.WSClient
 
-class PromoLandingPage @Inject()(override val wsClient: WSClient) extends Controller with OAuthActions {
-
-  private val tpBackend = TouchpointBackend.Normal
-  private val catalog = tpBackend.catalogService.unsafeCatalog
-
-  private val digitalPackRatePlanIds = catalog.digipack.plans.map(_.id).toSet
-  private val allPaperPackages = catalog.delivery.list ++ catalog.voucher.list
-  private val paperPlusPackageRatePlanIds = allPaperPackages.filter(_.charges.benefits.list.contains(Digipack)).map(_.id).toSet
-  private val paperOnlyPackageRatePlanIds = allPaperPackages.filterNot(_.charges.benefits.list.contains(Digipack)).map(_.id).toSet
-  private val guardianWeeklyRatePlanIds = catalog.weekly.zoneA.plans.map(_.id).toSet ++ catalog.weekly.zoneC.plans.map(_.id).toSet
-
-  private def isActive(promotion: AnyPromotion): Boolean = {
-    val isTest = tpBackend.environmentName != "PROD"
-    isTest || (
-      promotion.starts.isBeforeNow &&
-      !promotion.expires.exists(_.isBeforeNow) &&
-      promotion.appliesTo.productRatePlanIds.nonEmpty &&
-      promotion.appliesTo.countries.nonEmpty
-    )
-  }
+import scala.concurrent.Future
+import scalaz.OptionT
+import scalaz.std.scalaFuture._
+
+class PromoLandingPage (override val wsClient: WSClient, tpBackend: TouchpointBackend) extends Controller with OAuthActions {
+
+  private val catalogPromoLandingPage: Future[CatalogPromoLandingPage] = tpBackend.catalogService.catalog.map(_.map(c => new CatalogPromoLandingPage(c)).valueOr(e => throw new IllegalStateException(s"$e while getting catalog")))
+
+  class CatalogPromoLandingPage(val catalog: Catalog) {
```

Comment:
this hasn't changed as much as it looks, I just moved all the things that relied on a catalog into this nested class so we could avoid changing the methods at all, but other methods that don't rely on a catalog would be unchanged so wouldn't end up returning a future.
Unfrotauntely due to all the code being "text" it thinks everything has changed anyway, slightly negating my effort...

In reality these classes should be separate, a lot of the classes I worked on were not  very cohesive, but that's not a job for this PR!